src/dataServices/dynamoDbDataService.ts

```diff
@@ -103,6 +103,11 @@ export class DynamoDbDataService implements Persistence, BulkDataAccess {
     }
 
     private async createResourceWithId(resourceType: string, resource: any, resourceId: string) {
+        const regex = new RegExp('^[a-zA-Z0-9-.]{1,64}$');
+        if (!regex.test(resourceId)) {
+            throw new Error(`Resource creation failed, id ${resourceId} is not valid`);

```

Comment:
In this case -- it would be more accurate to throw a `InvalidResourceError`; since this will only fail if folks are using `updateCreate`: https://github.com/awslabs/fhir-works-on-aws-interface/blob/mainline/src/errors/InvalidResourceError.ts

FYI this will result in a 400 bad request being returned: https://github.com/awslabs/fhir-works-on-aws-routing/blob/mainline/src/router/routes/errorHandling.ts#L31